agent/agent-profiler/agent-alerting/src/main/java/com/microsoft/applicationinsights/alerting/analysis/data/TelemetryDataPoint.java

```diff
@@ -31,17 +31,23 @@ public class TelemetryDataPoint implements Comparable<TelemetryDataPoint> {
   private final AlertMetricType type;
   private final Instant time;
   private final double value;
+  private final String name;
 
-  public TelemetryDataPoint(AlertMetricType type, Instant time, double value) {
+  public TelemetryDataPoint(AlertMetricType type, Instant time, String name, double value) {
```

Comment:
I pushed a commit to turn this class into `@AutoValue` (let me know if it needed to implement Comparable)

Reply:
No, should be fine